SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -516,21 +516,44 @@ bool SpatialCommandUtils::FetchRuntimeBinary(const FString& RuntimeVersion, cons
 {
 	FString RuntimePath =
 		FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, SpatialGDKServicesConstants::RuntimePackageName, RuntimeVersion);
-	return FetchPackageBinary(RuntimeVersion, SpatialGDKServicesConstants::RuntimeExe, SpatialGDKServicesConstants::RuntimePackageName,
-							  RuntimePath, bIsRunningInChina, true);
+	return FetchPackageBinaryWithRetries(RuntimeVersion, SpatialGDKServicesConstants::RuntimeExe,
+										 SpatialGDKServicesConstants::RuntimePackageName, RuntimePath, bIsRunningInChina, true);
 }
 
 bool SpatialCommandUtils::FetchInspectorBinary(const FString& InspectorVersion, const bool bIsRunningInChina)
 {
 	FString InspectorPath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, SpatialGDKServicesConstants::InspectorPackageName,
 											InspectorVersion, SpatialGDKServicesConstants::InspectorExe);
-	return FetchPackageBinary(InspectorVersion, SpatialGDKServicesConstants::InspectorExe,
-							  SpatialGDKServicesConstants::InspectorPackageName, InspectorPath, bIsRunningInChina, false);
+	return FetchPackageBinaryWithRetries(InspectorVersion, SpatialGDKServicesConstants::InspectorExe,
+										 SpatialGDKServicesConstants::InspectorPackageName, InspectorPath, bIsRunningInChina, false);
+}
+
+bool SpatialCommandUtils::FetchPackageBinaryWithRetries(const FString& PackageVersion, const FString& PackageExe,
+														const FString& PackageName, const FString& SaveLocation,
+														const bool bIsRunningInChina, const bool bUnzip, int32 Retries /*= 3*/)
+{
+	while (Retries > 0 && !FetchPackageBinary(PackageVersion, PackageExe, PackageName, SaveLocation, bIsRunningInChina, bUnzip))
+	{
+		Retries--;
+		if (Retries == 0)
+		{
+			UE_LOG(LogSpatialCommandUtils, Error, TEXT("Giving up trying to fetch %s binary"), *PackageName);
+		}
+		else
+		{
+			UE_LOG(LogSpatialCommandUtils, Error, TEXT("Retrying to fetch %s binary"), *PackageName);
```

Comment:
Don't think this should be an error as this will cause NWX to have issues. Let's make it a log for now

Reply:
Maybe a warn?

Reply:
I guess log is appropriate actually 👍